src/warpedview.jl

```diff
@@ -0,0 +1,59 @@
+immutable WarpedView{T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I} <: AbstractArray{T,N}
+    parent::A
+    transform::F1
+    transform_inv::F2
+    indices::I
+
+    function (::Type{WarpedView{T,N,A,F1,F2,I}}){T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I}(
+            parent::A, tform::F1, tinv::F2, indices::I)
+        @assert eltype(parent) == T
+        new{T,N,A,F1,F2,I}(parent, tform, tinv, indices)
+    end
+end
+
+function WarpedView{T,N,F<:Transformation}(inner::WarpedView{T,N}, tform_in::F)
+    tform = compose(tform_in, inner.transform)
+    tinv = inv(tform)
+    etp = inner.parent
+    inds = autorange(etp, tform)
+    WarpedView{T,N,typeof(etp),typeof(tform),typeof(tinv),typeof(inds)}(etp, tform, tinv, inds)
+end
+
+function WarpedView{T,N}(parent::AbstractArray{T,N}, args...)
+    itp = Interpolations.BSplineInterpolation{T,N,typeof(parent),Interpolations.BSpline{Interpolations.Linear},OnGrid,0}(parent)
```

Comment:
The point of `prefilter` is that interpolation beyond `Linear` order requires a different coefficient matrix than the input values. For example:
```julia
julia> using Interpolations

julia> a = 1:5
1:5

julia> ai = interpolate(a, BSpline(Quadratic(Flat())), OnCell())
5-element Interpolations.BSplineInterpolation{Float64,1,Array{Float64,1},Interpolations.BSpline{Interpolations.Quadratic{Interpolations.Flat}},Interpolations.OnCell,1}:
 1.0
 2.0
 3.0
 4.0
 5.0

julia> ai.coefs
7-element Array{Float64,1}:
 0.853659
 0.853659
 2.02439 
 3.0     
 3.97561 
 5.14634 
 5.14634 
```
What `prefilter` does is essentially "invert" the computation (at the grid points) that interpolation will perform. For quadratic interpolation, the output for on-grid interpolation is given by
```julia
out[i] = (1/8)*coef[i-1] + (3/4)*coef[i] + (1/8)*coef[i+1]
```
and so the coefficients are calculated by inverting a tridiagonal system of equations.